subprojects/resources-s3/src/main/java/org/gradle/internal/resource/transport/aws/s3/S3ConnectorFactory.java

```diff
@@ -64,6 +64,6 @@ public ExternalResourceConnector createResourceConnector(ResourceConnectorSpecif
             }
         }
 
-        throw new IllegalStateException("S3 resource should either specify AwsIamAutentication or provide some AwsCredentials.");
```

Comment:
Why did you change that back to the old exception? Changing the test would probably have been better.